src/geowombat/core/api.py

```diff
@@ -1187,26 +1190,36 @@ def apply(
 
         if outfile is not None:
             profile = {
-                'count': apply_func_.count,
-                'width': self.width,
-                'height': self.height,
-                'crs': self.crs,
-                'transform': self.transform,
-                'driver': 'GTiff',
-                'dtype': apply_func_.dtype,
-                'compress': apply_func_.compress,
-                'sharing': False,
-                'tiled': True,
-                'nodata': self.nodata,
-                'blockxsize': self.blockxsize,
-                'blockysize': self.blockysize,
+                "count": apply_func_.count,
+                "width": self.width,
+                "height": self.height,
+                "crs": self.crs,
+                "transform": self.transform,
+                "driver": "GTiff",
+                "dtype": apply_func_.dtype,
+                "compress": apply_func_.compress,
+                "sharing": False,
+                "tiled": True,
+                "nodata": self.nodata,
+                "blockxsize": self.blockxsize,
+                "blockysize": self.blockysize,
+                **kwargs,
             }
 
+            # check for bigtiff config
+            if config["with_config"]:
+                bigtiff = config["bigtiff"].upper()
+                assert bigtiff in (

```

Comment:
Looks good, except can we make this raise an exception?
```python
if bigtiff not in ("YES", "NO", "IF_NEEDED", "IF_SAFER",):
    raise NameError("The GDAL BIGTIFF must be one of 'YES', 'NO', 'IF_NEEDED', 'IF_SAFER'. See https://gdal.org/drivers/raster/gtiff.html")
```

Reply:
Ok updated